src/shared/hooks/__tests__/useBasket.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+
+import { Namespace } from '../../types/namespaces';
+
+import useBasket from '../useBasket';
+
+describe('useBasket hook', () => {
+  afterEach(() => {
+    window.localStorage.clear();
+  });
+
+  test('get value, basic, first time', async () => {
+    const { result } = renderHook(() => useBasket());
+
+    expect(result.current[0]).toEqual({
+      [Namespace.uniprotkb]: new Set(),
+      [Namespace.uniref]: new Set(),
+      [Namespace.uniparc]: new Set(),
+    });
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    expect(JSON.parse(window.localStorage.getItem('basket')!)).toEqual({
+      [Namespace.uniprotkb]: [],
+      [Namespace.uniref]: [],
+      [Namespace.uniparc]: [],
+    });
+  });
+
+  test('get value, basic, first time, already saved', () => {
+    window.localStorage.setItem(
+      'basket',
+      JSON.stringify({
+        [Namespace.uniprotkb]: ['P05067'],
+        [Namespace.uniref]: [],
+        [Namespace.uniparc]: ['UPI0000000001', 'UPI0000000002'],
+      })
+    );
+    const { result } = renderHook(() => useBasket());
+
+    expect(result.current[0]).toEqual({
+      [Namespace.uniprotkb]: new Set(['P05067']),
+      [Namespace.uniref]: new Set(),
+      [Namespace.uniparc]: new Set(['UPI0000000001', 'UPI0000000002']),
+    });
+  });
+
+  test('set value, basic, already saved', async () => {
+    window.localStorage.setItem(
+      'basket',
+      JSON.stringify({
+        [Namespace.uniprotkb]: ['P05067'],
+        [Namespace.uniref]: [],
+        [Namespace.uniparc]: ['UPI0000000001', 'UPI0000000002'],
+      })
+    );
+    const { result } = renderHook(() => useBasket());
+
+    act(() =>
+      result.current[1]({
```

Comment:
Confused by this. why `current[1]`?

Reply:
current is the result of the hook. Since it's a tuple of `[basket, setBasket]`, `current[1]` is `setBasket`.
All the tests in this file are a copy of the ones from useLocalStorage but adapted a bit, so it's the same logic there.